Automattic/wp-calypso
# Plugins update manager: Schedule form, plugin selection

Related to https://github.com/Automattic/dotcom-forge/issues/5544
Closes https://github.com/Automattic/dotcom-forge/issues/5647

## Proposed Changes

* Implemented plugin selection with search and select all options
* Maximum selectable plugins are set to 10
* Endpoint used for plugin fetching `v1.2/sites/{SITE_ID}/plugins`

NOTE:
- This PR only contains plugin management inside the ScheduleForm component.
- Empty search result could be covered in the next PRs (I have to check wit designer)

## Testing Instructions



* Use GH direct link or local calypso
* Go to `/plugins/update-manager/create/{ATOMIC_SITE_SLUG}`
* Check the select all option (if the site has less or equal to 10 plugins)
* Check search options
* Check the plugin selection

| Maximum selectable plugins | Select all (for testing purposes, there are 11 plugins max) |
|--------|--------|
| ![Screen Capture on 2024-02-26 at 13-17-37](https://github.com/Automattic/wp-calypso/assets/1241413/2b8bc946-07e3-4114-b517-e3fd1e2c6e0a) | ![Screen Capture on 2024-02-26 at 13-19-50](https://github.com/Automattic/wp-calypso/assets/1241413/cd285ba5-8dfe-4e34-96f1-18fbc7afb6d7) | 





## Pre-merge Checklist
